trustmanager/keystore.go

```diff
@@ -40,7 +42,46 @@ func NewKeyFileStore(baseDir string, p notary.PassRetriever) (*GenericKeyStore,
 	if err != nil {
 		return nil, err
 	}
-	return NewGenericKeyStore(fileStore, p), nil
+	migrateTo0Dot4(fileStore)
+	store := NewGenericKeyStore(fileStore, p)
+	return store, nil
+}
+
+func migrateTo0Dot4(s Storage) {
+	for _, file := range s.ListFiles() {
+		keyID := filepath.Base(file)
+		fileDir := filepath.Dir(file)
+		d, _ := s.Get(file)
+		block, _ := pem.Decode(d)
+		if block == nil {
+			logrus.Warn("Key data for ", file, " may have been tampered with/ is invalid. The key has not been migrated and may not be available")
+			continue
+		}
+		if strings.HasPrefix(fileDir, notary.RootKeysSubdir) {
+			fileDir = strings.TrimPrefix(fileDir, notary.RootKeysSubdir)
+			if strings.Contains(keyID, "_") {
+				role := strings.Split(keyID, "_")[1]
+				keyID = strings.TrimSuffix(keyID, "_"+role)
+				block.Headers["role"] = role
+			}
+		} else if strings.HasPrefix(fileDir, notary.NonRootKeysSubdir) {
+			fileDir = strings.TrimPrefix(fileDir, notary.NonRootKeysSubdir)
+			if fileDir != "" {
+				block.Headers["gun"] = fileDir[1:]
+			}
+			if strings.Contains(keyID, "_") {
+				role := strings.Split(keyID, "_")[1]
+				keyID = strings.TrimSuffix(keyID, "_"+role)
+				block.Headers["role"] = role
+			}
+		}
+		var keyPEM bytes.Buffer
+		_ = pem.Encode(&keyPEM, block)
```

Comment:
Do we want to comment here that we are ignore the error return value because since block came from decoding the PEM bytes in the first place, and all we're doing is adding some headers, it shouldn't error in the first place?